gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AgentDirectoryMapper.java

```diff
@@ -0,0 +1,255 @@
+    private static final String ORG_META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Organization-1";
+    private static final String PRACT_ROLE_META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC"
+        + "-PractitionerRole-1";
+    private static final String ORG_IDENTIFIER_SYSTEM = "https://fhir.nhs.uk/Id/ods-organization-code";
+    private static final String ORG_PREFIX = "Organization/";
+    private static final String PRACT_PREFIX = "Practitioner/";
+    private static final String ORG_ID_SUFFIX = "-ORG";
+    private static final String PRACT_ROLE_SUFFIX = "-PR";
+    private static final String ORG_ROOT = "2.16.840.1.113883.2.1.4.3";
+    private static final String UNKNOWN = "Unknown";
+    private static final String WORK_PLACE = "WP";
+    private static final int TELECOM_RANK = 1;
+    private static final int TEL_PREFIX_INT = 4;
+
+    public List mapAgentDirectory(RCMRMT030101UK04AgentDirectory agentDirectory) {
+        var partList = agentDirectory.getPart();
+        if (!CollectionUtils.isEmpty(partList)) {
+            return partList.stream()
+                .map(RCMRMT030101UK04Part::getAgent)
+                .map(this::mapAgent)
+                .flatMap(List::stream)
+                .toList();
+        }
+
+        return null;
+    }
+
+    private List mapAgent(RCCTMT120101UK01Agent agent) {
+        var agentResourceList = new ArrayList<>();
+        var agentPerson = agent.getAgentPerson();
+        var agentOrganization = agent.getAgentOrganization();
+        var representedOrganization = agent.getRepresentedOrganization();
+        var resourceId = agent.getId().get(0).getRoot();
+
+        if (agentPerson != null && representedOrganization != null) {
+            agentResourceList.add(createPractitioner(agentPerson, resourceId));
+            agentResourceList.add(createRepresentedOrganization(representedOrganization, resourceId));
+            agentResourceList.add(createPractitionerRole(resourceId, agent.getCode()));
+        } else if (agentPerson != null && agentOrganization == null) {
+            agentResourceList.add(createPractitioner(agentPerson, resourceId));
+        } else if (agentPerson == null && agentOrganization != null) {
+            agentResourceList.add(createAgentOrganization(agentOrganization, resourceId, agent.getCode()));
+        }
+
+        return agentResourceList;
+    }
+
+    private Practitioner createPractitioner(RCCTMT120101UK01Person agentPerson, String id) {
+        var practitioner = new Practitioner();
+
+        practitioner.setId(id);
+        practitioner.getMeta().getProfile().add(new UriType(PRACT_META_PROFILE));
+        practitioner.setName(getPractitionerName(agentPerson.getName()));
+
+        return practitioner;
+    }
+
+    private List<HumanName> getPractitionerName(PN name) {
+        var nameList = new ArrayList();
```

Comment:
```suggestion
        var nameList = new ArrayList<HumanName>();
```
same here in the declaration of ArrayList?